src/api-collector/main.go

```diff
@@ -0,0 +1,216 @@
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/google/go-github/v61/github"
+	"golang.org/x/oauth2"
+	"gopkg.in/yaml.v2"
+)
+
+const API_DOCS_REPO = "api-hub"
+
+type assetInfo struct {
+	id   int64
+	name string
+}
+
+type OpenAPIInfo struct {
+	Version string `yaml:"version"`
+	Title   string `yaml:"title"`
+}
+type OpenAPISpec struct {
+	OpenAPI string      `yaml:"openapi"`
+	Info    OpenAPIInfo `yaml:"info"`
+}
+
+func main() {
+	owner, token := getArgs()
+	ctx := context.Background()
+	client := getAuthenticatedClient(ctx, token)
+	repos, err := getOrgRepos(ctx, owner, client)
+	if err != nil {
+		log.Fatalf("Error fetching repos: %s", err)
+	}
+	for _, repo := range repos {
+		log.Println("Scanning repo ", *repo.Name)
+		specAssets := getAPISpecAssets(ctx, client, owner, *repo.Name)
+		if specAssets == nil {
+			log.Println("\t- No OpenAPI specs found")
+			continue
+		}
+		downloadedSpecs := downloadAPISpecs(ctx, client, owner, *repo.Name, specAssets)
+		commitDownloadedSpec(ctx, client, owner, API_DOCS_REPO, downloadedSpecs)
+	}
+}
+
+func getArgs() (string, string) {
+	owner := flag.String("owner", "", "Specify GitHub User or Organization")
+	token := flag.String("token", "", "Specify GitHub Token")
+	flag.Parse()
+
+	if *owner == "" || *token == "" {
+		log.Fatalln("Missing required arguments, please specify -owner [owner] and -token [token]")
+	}
+	return *owner, *token
+}
+
+func getAuthenticatedClient(ctx context.Context, gitToken string) *github.Client {
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: gitToken},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+
+	return github.NewClient(tc)
+}
+
+func getOrgRepos(ctx context.Context, gitOwner string, client *github.Client) ([]*github.Repository, error) {
+	opt := &github.RepositoryListByOrgOptions{
+		ListOptions: github.ListOptions{},
+	}
+
+	var allRepos []*github.Repository
+
+	for {
+		repos, response, err := client.Repositories.ListByOrg(ctx, gitOwner, opt)
+		if err != nil {
+			return nil, err
+		}
+		allRepos = append(allRepos, repos...)
+		if response.NextPage == 0 {
+			break
+		}
+		opt.Page = response.NextPage
+	}
+
+	return allRepos, nil
+}
+
+func getAPISpecAssets(ctx context.Context, client *github.Client, owner string, repo string) []assetInfo {
+	var apiSpecs []assetInfo
+	release, _, err := client.Repositories.GetLatestRelease(ctx, owner, repo)
```

Comment:
what if as an alternative way to get the documentation, the url for a published openapi yaml file could be specified an url in the `.tractusx` file?

it will make this process more flexible, because the yaml could then be published either:
- in the releases file (it can be referenced with a url like: `https://github.com/eclipse-tractusx/<project-name>/releases/latest/download/openapi.yaml`)
- on github pages
- kept in the repository in the `docs` folder

it will be up to the single project to decide where to store it

Reply:
Hi @ndr-brt, thanks for your suggestion! I think I like the flexibility it gives to devs..